src/2-run-ipfs-on-a-vps/index.md

```diff
@@ -0,0 +1,79 @@
+---
+baseurl: ..
+template: tmpl/layouts/post.html
+breadcrumbs:
+  - {name: "2-run-ipfs-on-a-vps", link: "./" }
+id: 2-run-ipfs-on-a-vps
+date: 2015-11-02
+title: Run IPFS latest on a VPS
+author: Kyle Drake
+collection: posts
+---
+
+The best way to provide content using [IPFS](https://ipfs.io) is to run your own IPFS node. You can do this by running an IPFS node on your personal computer, but that will only work as long as your computer is running. For hosting content with a higher availability, you need to run an IPFS node in a datacenter. That ensures your content is always online and available to other nodes on the network.
+
+VPS instances provided by [Digital Ocean](https://www.digitalocean.com/), [Ramnode](http://ramnode.com/), [Linode](https://www.linode.com/), [Vultr](https://www.vultr.com/) and many other providers allow you to quickly setup your own Linux server with the reliability of a managed dedicated server without the full cost. This is a quick guide to setting up your own dedicated IPFS node on a VPS. We'll be using [Ubuntu](http://www.ubuntu.com/) 14.04LTS 64-bit for the example.
+
+First, some initial housekeeping I like to do. We're going to update the installed system to make sure that it has the latest updates. We're also going to turn on unattended updates, so that security patches can be done automatically without your intervention, and setup OpenNTP so the system clock is automatically updated:
+
+    apt-get update
+    apt-get upgrade
+    apt-get install openntpd unattended-upgrades unzip wget
+    dpkg-reconfigure unattended-upgrades
+
+Now you can download the latest build of IPFS from the [install page](https://ipfs.io/docs/install/). We'll be using Linux x86_64:
+
+    wget https://gobuilder.me/get/github.com/ipfs/go-ipfs/cmd/ipfs/ipfs_master_linux-amd64.zip
+    unzip ipfs_master_linux-amd64.zip
+    cp ipfs/ipfs /usr/local/bin/
+
+It's usually not a good idea to run a public-facing service as root. So we'll create a user account to run IPFS in and switch to it:
+
+    adduser ipfs
+    su ipfs
+
+First let's initialize the IPFS config:
+
+    ipfs init
+
+This installs a basic IPFS config file for you.
+
+Now we'll add a blocklist for local networks. IPFS works by actively seeking nearby nodes to connect to, which is a good thing for performance and availability. Unfortunately, many VPS providers incorrectly detect this as malware activity, so this is a temporary workaround you can use for now:
```

Comment:
explain a bit more --  (i wouldnt use something that was detected as malware without a good explanation.) -- like maybe:

> IPFS works by actively seeking nearby nodes to connect to, which is a good thing for performance and availability, particularly in home and office networks. This causes addresses in the networks to be dialed that may not be there. Unfortunately, some VPS providers _incorrectly_ classify this as suspicious activity, and some even have blocked nodes for doing so. To avoid this, let's do two things:
>
> ```sh
> # 1. disable mDNS discovery
> ipfs config --json Discovery.MDNS.Enabled false
>
> # 2. filter out local network addresses
> ipfs config --json Swarm.AddrFilters '[
> ```